crtclient/crtclient.go

```diff
@@ -15,29 +15,26 @@ limitations under the License.
 
 package crtclient
 
-// container runtime client interface implementations
-
-type Config struct {
-	// Config object parsed from a json styled config
-	V interface{}
-}
-
-type ContainerEpContext struct {
+// ContainerEPContext is used to manage the parameters for all
+// container+endpoint management operations.
+type ContainerEPContext struct {
 	NewContName    string
 	NewAttachUUID  string
 	CurrContName   string
 	CurrAttachUUID string
-	InterfaceId    string
-	IpAddress      string
+	InterfaceID    string
+	IPAddress      string
 	SubnetLen      uint
 	DefaultGw      string
 }
 
+// ContainerIf implementations are used to configure and manage container
+// interfaces.
 type ContainerIf interface {
-	Init(config *Config) error
+	Init(config interface{}) error
```

Comment:
Yep, you are right. `type Config interface{}` shall also work.

Initially when I had defined `core.Config` struct my understanding was that it will be more explicit to pass a struct containing json config as argument to the Init() routines of `drivers`. But thinking more having a type-def like `type Config interface{}` achieves the same by keeping the type of the config argument more explicit. Still I think we should use the same `core.Config` type here as well than defining a new type, may be we can make the change as part of separate PR or whatever works for you.